src/libdredd/src/mutation_replace_binary_operator.cc

```diff
@@ -203,31 +246,88 @@ std::string MutationReplaceBinaryOperator::GetFunctionName(
                                   ->getName(ast_context.getPrintingPolicy())
                                   .str());
 
+  if (!binary_operator_.isAssignmentOp()) {
+    clang::Expr::EvalResult eval_result;
+    bool rhs_is_int =
+        binary_operator_.getRHS()->EvaluateAsInt(eval_result, ast_context);
+    if (rhs_is_int && llvm::APSInt::isSameValue(eval_result.Val.getInt(),
+                                                llvm::APSInt::get(0))) {
+      result += "_rhs_zero";
+    } else if (rhs_is_int && llvm::APSInt::isSameValue(eval_result.Val.getInt(),
+                                                       llvm::APSInt::get(1))) {
+      result += "_rhs_one";
+    } else if (rhs_is_int && llvm::APSInt::isSameValue(eval_result.Val.getInt(),
+                                                       llvm::APSInt::get(-1))) {
+      result += "_rhs_minus_one";
+    }
+
+    bool lhs_is_int =
+        binary_operator_.getLHS()->EvaluateAsInt(eval_result, ast_context);
+    if (lhs_is_int && llvm::APSInt::isSameValue(eval_result.Val.getInt(),
+                                                llvm::APSInt::get(0))) {
+      result += "_lhs_zero";
+    } else if (lhs_is_int && llvm::APSInt::isSameValue(eval_result.Val.getInt(),
+                                                       llvm::APSInt::get(1))) {
+      result += "_lhs_one";
+    } else if (lhs_is_int && llvm::APSInt::isSameValue(eval_result.Val.getInt(),
+                                                       llvm::APSInt::get(-1))) {
+      result += "_lhs_minus_one";
+    }
+  }
+
   return result;
 }
 
 void MutationReplaceBinaryOperator::GenerateArgumentReplacement(
     const std::string& arg1_evaluated, const std::string& arg2_evaluated,
-    std::stringstream& new_function, int& mutant_offset) const {
+    clang::ASTContext& ast_context, std::stringstream& new_function,
+    int& mutant_offset) const {
   if (!binary_operator_.isAssignmentOp()) {
+    clang::Expr::EvalResult lhs_eval_result;
+    clang::Expr::EvalResult rhs_eval_result;
+    bool rhs_is_int =
+        binary_operator_.getRHS()->EvaluateAsInt(rhs_eval_result, ast_context);
```

Comment:
I suggest moving the declaration of rhs_eval_result, and the call to EvaluateAsInt for RHS, until right before the block of code where you use it.

In general it's good practice to put declarations close to their use, to prevent the possibility of accidentally referring to the wrong variable.

Reply:
Done.